packages/selected-network-controller/src/SelectedNetworkController.ts

```diff
@@ -136,6 +143,48 @@ export class SelectedNetworkController extends BaseController<
     });
     this.#registerMessageHandlers();
 
+    this.messagingSystem
+      .call('PermissionController:getSubjectNames')
+      .filter(
+        (domain) =>
+          this.state.domains[domain] === undefined &&
+          this.state.domains[domain] === undefined,
+      )
+      .forEach((domain) =>
+        this.setNetworkClientIdForDomain(
+          domain,
+          this.messagingSystem.call('NetworkController:getState')
+            .selectedNetworkClientId,
+        ),
+      );
+
+    this.messagingSystem.subscribe(
+      'PermissionController:stateChange',
+      (_, patches) => {
+        patches.forEach(({ op, path }) => {
+          const isChangingSubject =
+            path[0] === 'subjects' && path[1] !== undefined;
```

Comment:
What is `path[1] !== undefined` doing here?

Reply:
path[1] could be undefined if you were to say set the entire 'subjects' object. Not something we really expect, but this also makes typescript happy.